internal/core/domain/withdrawal_repository.go

```diff
@@ -4,10 +4,8 @@ import "context"
 
 type WithdrawalRepository interface {
 	AddWithdrawal(ctx context.Context, withdrawal Withdrawal) error
-	ListWithdrawalsForAccountIdAndPage(
-		ctx context.Context,
-		accountIndex int,
-		page Page,
+	ListWithdrawalsForAccountId(
+		ctx context.Context, accountIndex int, page *Page,
 	) ([]Withdrawal, error)
-	ListAllWithdrawals(ctx context.Context) ([]Withdrawal, error)
+	ListAllWithdrawals(ctx context.Context, page *Page) ([]Withdrawal, error)
```

Comment:
Based on name all objects should be retrieved and therefor Page object shouldnot be passed as arg.

Reply:
you're right about the puprose of the method, but still, all withdrawals can be returned in a paginated way.

Reply:
same answer applies for comments below

Reply:
I think we could have a little repetition but more clear

- ListAllWithdrawals
- ListPaginatedWithdrawals

Reply:
Do you think it's more clear this way? consider also that this repetition would be needed for all other List/GetAll methods of the remaining repositories.

Another option could be to add comments (that will be part of the code documentation, therefore) describing the intended behavior for these methods.